src/hanabi.sml

```diff
@@ -330,19 +340,21 @@ struct
          hands = tl (#hands s),
          log = getLog (#log s) (length ps),
          turns = #turns s,
+         turnNumber = #turnNumber s,
          inPlay = #inPlay s,
          inDiscard = #inDiscard s}
       val s' = if illegalMove (act,s)
                then raise Fail ("Illegal action: " ^ actionToString act ^ "\n")
                else enact (act,s)
     in
-      if (trace (act,s'); gameOver s')
+      if (trace (act,s); gameOver s')
       then s'
```

Comment:
If we're printing before the turn instead of after, then when the game ends shouldn't we `trace (act,s')` so the ending state is printed once?